demos/vc_issuer/tests/issue_credential.rs

```diff
@@ -163,36 +163,41 @@ mod api {
 /// Verifies that the consent message can be requested.
 #[test]
 fn should_return_vc_consent_message() {
+    let test_cases = [
+        ("en-US", "en", "# DFINITY Foundation Employment Credential"),
+        ("de-DE", "de", "# Beschäftigungsausweis DFINITY Stiftung"),
+        ("ja-JP", "en", "# DFINITY Foundation Employment Credential"), // test fallback language
+    ];
     let env = env();
     let canister_id = install_canister(&env, VC_ISSUER_WASM.clone());
 
-    let mut args = HashMap::new();
-    args.insert(
-        "employerName".to_string(),
-        ArgumentValue::String("DFINITY Foundation".to_string()),
-    );
-    let consent_message_request = Icrc21VcConsentMessageRequest {
-        credential_spec: CredentialSpec {
-            credential_name: "VerifiedEmployee".to_string(),
-            arguments: Some(args),
-        },
-        preferences: Icrc21ConsentPreferences {
-            language: "en-US".to_string(),
-        },
-    };
+    for (requested_language, actual_language, consent_message_snippet) in test_cases {
+        let mut args = HashMap::new();
+        args.insert(
+            "employerName".to_string(),
+            ArgumentValue::String("DFINITY Foundation".to_string()),
+        );
+        let consent_message_request = Icrc21VcConsentMessageRequest {
+            credential_spec: CredentialSpec {
+                credential_name: "VerifiedEmployee".to_string(),
+                arguments: Some(args),
+            },
+            preferences: Icrc21ConsentPreferences {
+                language: requested_language.to_string(),
+            },
+        };
 
-    let response =
-        api::vc_consent_message(&env, canister_id, principal_1(), &consent_message_request)
-            .expect("API call failed")
-            .expect("Got 'None' from vc_consent_message");
-    assert_matches!(response, Icrc21ConsentMessageResponse::Ok(_));
-    if let Icrc21ConsentMessageResponse::Ok(info) = response {
-        assert_eq!(info.language, "en-US");
-        assert!(info
-            .consent_message
-            .contains("Issue credential 'VerifiedEmployee'"));
-        assert!(info.consent_message.contains("employerName"));
-        assert!(info.consent_message.contains("DFINITY Foundation"));
+        let response =
+            api::vc_consent_message(&env, canister_id, principal_1(), &consent_message_request)
+                .expect("API call failed")
+                .expect("Got 'None' from vc_consent_message");
+        assert_matches!(response, Icrc21ConsentMessageResponse::Ok(_));
+        if let Icrc21ConsentMessageResponse::Ok(info) = response {
+            assert_eq!(info.language, actual_language);
+            assert!(info
+                .consent_message
+                .starts_with(consent_message_snippet));
+        }
```

Comment:
not related to this PR, but would it make sense to panic if the response is not what we expect? `else { panic!(...) }`